### DISCUSSION TITLE
HASPmota on Lanbon L8

### DISCUSSION THREAD

**Discussion by waltkind** [arendst/Tasmota#20180] [Support request] [answered]
While playing aroung with openHASP on a new Lanbon L8 I came around HASPmota. Having some devices with tasmota working perfect in my home I tried HASPmota.
For me it seems to be the perfect combination of disigning the display and the programming of tasmota.

The minimal requirements say "Currently PSRAM is strongly recommended to run HASPmota" but not that it is necessary.
The console tells: "Error: OpenHASP demo requires PSRAM".

Did someone get the combination to work? I can reach the UI, but the display stays dark.

**Comment by sfromis** [comment]
If your setup for the display is done right, you should briefly see a Tasmota logo on the panel at boot time.

**Reply by waltkind** [reply]
At boot time I see that logo, but nothing else. 
In https://tasmota.github.io/docs/LVGL/ there are shown further screens which don't appear.

**Reply by sfromis** [reply, marked as answer]
You did remember to power on the display, like via the Power command, or the extra toggle appearing on the front page?

Anyway, here's a small test `pages.jsonl`, which presents a clock and date, using a built-in font. I'm using a larger version (with more things on the screen) on another device, with another font loaded.
```
{"page":0,"id":0,"bg_color":"#000000","text_color":"#FFFFFF"}
{"id":1,"obj":"lv_clock","x":0,"y":0,"w":240,"h":96,"align":1,"text":"","radius":0,"border_side":0,"text_font":"montserrat-20"}
{"page":1,"id":0,"bg_color":"#000000","text_color":"#FFFFFF"}
{"id":1,"obj":"label","x":0,"y":112,"w":240,"align":1,"text":"Booted","text_rule":"Time#Minute","text_rule_format":"%s","text_rule_formula":"tasmota.strftime('%Y-%m-%d %a', tasmota.rtc()['local'])","text_font":"montserrat-20"}
```
I'd expect this to be workable without PSRAM.

**Comment by waltkind** [comment]
Thanks for the moment. I'll try this perhaps tomorrow when I have the time.

**Reply by waltkind** [reply]
Thanks sfromis for pushing me in the right direction.
The steps in https://tasmota.github.io/docs/HASPmota/#quick-tour have not been clear to me and I have stopped after the haspmota_demo.tapp didn't work for me because of lacking PSRAM.
When I went further, creating your pages.jsonl and the autoexec.be everything was fine.

During booting I see still two errors: _File 'mcp23x.dat' not found_  and _File '.drvset003' not found_
It seems to be no problem and searching the net shows other have the same in their logs, but I found no solution for that.

Now when I tried my pages.jsonl from openHASP I realized that there has to be some more tweaking to get the same results. But that should not be too much work.

**Reply by sfromis** [reply]
Those two non-existing files are not "missing", as long as you are not even trying to use the associated features. Not really error messages, just "less interesting information".
